web/obs/src/components/chat/ChatItemBox.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+  <TextMessage
+    v-if="message.type === types.MessageType.TEXT"
+    class="style-scope yt-live-chat-item-list-renderer"
+    :time="message.time"
+    :avatarUrl="message.avatarUrl"
+    :authorName="message.authorName"
+    :authorType="message.authorType"
+    :privilegeType="message.privilegeType"
+    :contentParts="getShowContentParts(message)" />
+  <PaidMessage
+    v-else-if="message.type === types.MessageType.GIFT"
+    class="style-scope yt-live-chat-item-list-renderer"
+    :time="message.time"
+    :avatarUrl="message.avatarUrl"
+    :authorName="getShowAuthorName(message)"
+    :price="message.price"
+    :priceText="message.price <= 0 ? getGiftShowNameAndNum(message) : ''"
+    :content="
+      message.price <= 0 ? '' : getGiftShowContent(message, showGiftName)
+    " />
+  <MembershipItem
+    v-else-if="message.type === types.MessageType.MEMBER"
+    class="style-scope yt-live-chat-item-list-renderer"
+    :time="message.time"
+    :avatarUrl="message.avatarUrl"
+    :authorName="getShowAuthorName(message)"
+    :privilegeType="message.privilegeType"
+    :title="message.title" />
+  <PaidMessage
+    v-else-if="message.type === types.MessageType.SUPER_CHAT"
+    class="style-scope yt-live-chat-item-list-renderer"
+    :time="message.time"
+    :avatarUrl="message.avatarUrl"
+    :authorName="getShowAuthorName(message)"
+    :price="message.price"
+    :priceText="''"
+    :content="getShowContent(message)" />
+</template>
+
+<script lang="ts" setup>
+import * as types from '../../types/ChatMessageType'
+import {
+  getGiftShowContent,
+  getGiftShowNameAndNum,
+  getShowAuthorName,
+  getShowContent,
+  getShowContentParts,
+} from './constants'
+import MembershipItem from './MembershipItem.vue'
+import PaidMessage from './PaidMessage.vue'
+import TextMessage from './TextMessage.vue'
+
+withDefaults(
+  defineProps<{
+    message: types.AnyDisplayMessage
+    showGiftName: boolean
+  }>(),
+  {
+    showGiftName: false,
+  },
+)
```

Comment:
这里没必要用 withDefaults
null undefined 都可以当 false 用
```suggestion
defineProps<{
  message: types.AnyDisplayMessage
  showGiftName?: boolean
}>()
```

Reply:
L20 附近的`getGiftShowContent(message, showGiftName)`不许传 null。不然像 c# 一样 fallback：`showGiftName ?? false`？